### DISCUSSION TITLE
Adding primaryKey to Models

### DISCUSSION THREAD

**Discussion by Ifnir** [getcandy/candy-api#396] [Ideas]
Atm I have trouble updating a variant through a relation.
```

$product = Product::where('id', '=', $productitem->Id)->first();
$productVariant[] = new ProductVariant([
'Price' => 7700]);

if( $product) $product->variants()->first()->save($productVariant);
```
This example code above returns "True", but doesn't update anything.
What I researched so far for why it only return "True" is, that there isn't really a explanation except primary key in model is being overwritten

So can we test if adding
`
protected $primaryKey = "id"
`
to "ProductVariant" model fixes the issue?

**Comment by glennjacobs** [comment]
Take a look at this as reference https://github.com/getcandy/candy-api/blob/master/src/Core/Products/Services/ProductVariantService.php

**Reply by Ifnir** [reply]
Thanks for reference, I found a solution.
